client-core/src/service/wallet_state_service.rs

```diff
@@ -192,11 +230,10 @@ impl Default for WalletState {
 
 impl WalletState {
     /// Applies memento to wallet state
-    fn apply_memento(&mut self, memento: &WalletStateMemento) -> Result<()> {
+    pub fn apply_memento(&mut self, memento: &WalletStateMemento) -> Result<()> {
```

Comment:
I guess this is now public as `WalletStateMemento` can be created outside `WalletState`? not sure if intended

Reply:
Yes, before this, all the operations on `WalletState` go through `WalletStateService`, now outside code will access `WalletState` directly.